auth/user.go

```diff
@@ -75,6 +75,7 @@ func (auth *Authenticator) defaultGuestUser() User {
 		auth: auth,
 	}
 	user.Channels_ = user.ExplicitChannels_.Copy()
+	user.Channels_.AddChannel(ch.DocumentStarChannel, 1)
```

Comment:
Feels cleaner to have `defaultGuestUser` not contain _any_ channel information by default, and have it be calculated on first auth, like how other users get access to the star channel.

That allows the `ch.DocumentStarChannel` assignment to remain in one place (`(Authenticator) rebuildChannels()`).